File: src/main/java/ludwig/ide/SignatureEditor.java
package ludwig.ide;

import javafx.scene.layout.VBox;
import ludwig.model.NamedNode;
import ludwig.model.Node;
import ludwig.model.OverrideNode;
import ludwig.model.VariableNode;
import ludwig.utils.NodeUtils;
import ludwig.workspace.Environment;


public class SignatureEditor extends VBox {
    private final Environment environment;
    private Node<?> node;

    public SignatureEditor(Environment environment) {
        this.environment = environment;
    }

    public void setNode(Node<?> node) {
        this.node = node;

        getChildren().clear();

        NamedNode<?> decl = (node instanceof OverrideNode) ? NodeUtils.INSTANCE.declaration((OverrideNode) node) : (NamedNode) node;
        if (node == null) {
            return;
        }

        getChildren().add(new SignatureItemEditor(environment, decl));

        for (Node n : decl.children()) {
            if (!(n instanceof VariableNode)) {
                break;
            }
            getChildren().add(new SignatureItemEditor(environment, (NamedNode<?>) n));
        }

    }
}

File: src/main/java/ludwig/ide/SignatureEditor.kt
package ludwig.ide

import javafx.scene.layout.VBox
import ludwig.model.NamedNode
import ludwig.model.Node
import ludwig.model.OverrideNode
import ludwig.model.VariableNode
import ludwig.utils.NodeUtils
import ludwig.workspace.Environment


class SignatureEditor(private val environment: Environment) : VBox() {
    private var node: Node<*>? = null

    fun setNode(node: Node<*>?) {
        this.node = node

        children.clear()

        val decl = if (node is OverrideNode) NodeUtils.declaration((node as OverrideNode?)!!) else node as NamedNode<*>?
        if (node == null) {
            return
        }

        children.add(SignatureItemEditor(environment, decl!!))

        for (n in decl.children) {
            if (n !is VariableNode) {
                break
            }
            children.add(SignatureItemEditor(environment, n as NamedNode<*>))
        }

    }
}
